### DISCUSSION TITLE
Benchmarks

### DISCUSSION THREAD

**Discussion by pfeatherstone** [davisking/dlib#2767] [Ideas]
Maybe it would be a good idea to use something like https://github.com/google/benchmark to benchmark dlib functionality on every release, or even every commit. 
We could add a Github Actions workflow which would run the benchmarks and store the results somewhere, maybe even updating some graphs or CSV tables.

**Comment by davisking** [comment]
I don't really want to add a dependency.  And it's easy to time stuff with just the standard library.

I doubt github actions gives reliable timing results.  Whenever I see stuff run on these kinds of cloud services the timings are all over the place.  I would just have something in dlib/tools if you want that runs a benchmark and you can then run it locally on a machine you know isn't running anything else and is a sensible reference.

**Comment by pfeatherstone** [comment]
https://github.com/davisking/dlib/discussions/2876
